core/services/ocr2/plugins/ocr2keeper/evm/registry.go

```diff
@@ -826,3 +840,52 @@ func (r *EvmRegistry) getUpkeepConfigs(ctx context.Context, ids []*big.Int) ([]a
 
 	return results, multiErr
 }
+
+// updateTriggerConfig gets invoked upon changes in the trigger config of an upkeep.
+func (r *EvmRegistry) updateTriggerConfig(id *big.Int, cfg []byte) {
+	uid := id.String()
+	switch getUpkeepType(ocr2keepers.UpkeepIdentifier(uid)) {
+	case logTrigger:
+		parsed, err := r.packerV21.UnpackLogTriggerConfig(string(cfg))
+		if err != nil {
+			r.lggr.Warnw("failed to unpack log upkeep config", "upkeepID", uid)
+			return // TODO: handle?
+		}
+		if err := r.logFilters.Register(id, LogTriggerConfig(parsed)); err != nil {
+			r.lggr.Warnw("failed to register log filter", "upkeepID", uid)
+			return // TODO: handle?
+		}
+		r.lggr.Debugw("registered log filter", "upkeepID", uid)
+	default:
+	}
+}
+
+type upkeepType int32
+
+const (
+	conditionTrigger upkeepType = iota
+	logTrigger
+	cronTrigger
+	readyTrigger
+)
+
```

Comment:
Can these consts and types live at the top of the file?

Reply:
They'll be moved to another file, I kept them here for easy merging